be/src/storage/persistent_index.cpp

```diff
@@ -3738,6 +3946,85 @@ Status PersistentIndex::_merge_compaction_internal(ImmutableIndexWriter* writer,
     return Status::OK();
 }
 
+size_t PersistentIndex::_get_tmp_l1_count() {
+    return _has_l1 ? _l1_vec.size() - 1 : _l1_vec.size();
+}
+
+// There are a few steps in async compaction:
+// 1. flush l0 to l1:
+//    a. if there is only one tmp-l1 file, move this tmp-l1 to l1 file.
+//    b. if there are > 2 tmp-l1 file, then merge l0 and tmp-l1 files to new l1 file.
+//    c. if there is only one l1 file, flush l0 to new l1 file.
+//    d. if there is not l1 file exists, flush l0 to l1 file.
+// 2. move old l1 to l2. only if old l1 exist
```

Comment:
Do you need to move the old l1 to l2, or just merge multiple l1 files to l2 when reach a limit?

Reply:
Move old l1 to l2 is better for apply latency.